pkg/river/parser/internal.go

```diff
@@ -683,34 +683,28 @@ func (p *parser) parseFieldList(until token.Token) []*ast.ObjectField {
 //
 //    Field = ( string | identifier ) "=" Expression
 func (p *parser) parseField() *ast.ObjectField {
-	if p.tok != token.STRING && p.tok != token.IDENT {
-		p.addErrorf("expected field name (string or identifier), got %s", p.tok)
-		p.advanceAny(fieldStarter)
-		return nil
-	}
+	var field ast.ObjectField
 
-	field := &ast.ObjectField{
-		Name: &ast.Ident{
+	if p.tok == token.STRING || p.tok == token.IDENT {
+		field.Name = &ast.Ident{
 			Name:    p.lit,
 			NamePos: p.pos,
-		},
-	}
-	if p.tok == token.STRING && len(p.lit) > 2 {
-		// The field name a string literal; unwrap the quotes.
-		field.Name.Name = p.lit[1 : len(p.lit)-1]
-		field.Quoted = true
+		}
+		if p.tok == token.STRING && len(p.lit) > 2 {
+			// The field name a string literal; unwrap the quotes.
+			field.Name.Name = p.lit[1 : len(p.lit)-1]
+			field.Quoted = true
+		}
+		p.next() // Consume field name
+	} else {
+		p.addErrorf("expected field name (string or identifier), got %s", p.tok)
+		p.advance(token.ASSIGN)
 	}
-	p.next() // Consume field name
 
 	p.expect(token.ASSIGN)
 
 	field.Value = p.ParseExpression()
-	return field
-}
-
-var fieldStarter = map[token.Token]struct{}{
```

Comment:
We no longer have a use for this map, so let's remove it. 

We _could_ use it to check that p.tok is in the `fieldStarter` set instead of checking directly for token.STRING and token.IDENT, but I'm not sure it makes any difference.